src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -263,6 +263,12 @@ object Erasure extends TypeTestsCasts{
           else
             cast(tree, pt)
       }
+
+
+    /** Is `tpe` a type which is a bottom type for Dotty, but not a bottom type for JVM */
+    def isNonJVMBottomType(tpe: Type)(implicit ctx: Context): Boolean = {
+      tpe.derivesFrom(ctx.definitions.NothingClass) || tpe.derivesFrom(ctx.definitions.NullClass)
+    }
   }
```

Comment:
No, I think it's OK to leave the two separate. Maybe rename the second to
`derivesFromBottomType`
to make clear that refinements and & / | count as well?


On Fri, Dec 18, 2015 at 6:32 PM, Dmitry Petrashko <notifications@github.com>
wrote:

> In src/dotty/tools/dotc/transform/Erasure.scala
> <https://github.com/lampepfl/dotty/pull/1002#discussion_r48048942>:
>
> > @@ -263,6 +263,12 @@ object Erasure extends TypeTestsCasts{
> >            else
> >              cast(tree, pt)
> >        }
> > +
> > +
> > +    /** Is `tpe` a type which is a bottom type for Dotty, but not a bottom type for JVM */
> > +    def isNonJVMBottomType(tpe: Type)(implicit ctx: Context): Boolean = {
> > +      tpe.derivesFrom(ctx.definitions.NothingClass) || tpe.derivesFrom(ctx.definitions.NullClass)
> > +    }
> >    }
>
> Should I update Definitions#isBottomType to account for Null | Nothing
> and Int & Nothing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lampepfl/dotty/pull/1002/files#r48048942>.
>



-- 
Martin Odersky
EPFL